src/type/types.cpp

```diff
@@ -549,6 +554,115 @@ std::ostream& operator<<(std::ostream& os, const StatementType& type) {
 }
 
 //===--------------------------------------------------------------------===//
+// QueryType - String Utilities
+//===--------------------------------------------------------------------===//
+std::string QueryTypeToString(QueryType query_type) {
+  switch(query_type) {
+    case QueryType::QUERY_BEGIN:return "BEGIN";
+    case QueryType::QUERY_COMMIT:return "COMMIT";
+    case QueryType::QUERY_ROLLBACK:return "ROLLBACK";
+    case QueryType::QUERY_CREATE_DB:return "CREATE DATABASE";
+    case QueryType::QUERY_CREATE_INDEX: return "CREATE INDEX";
+    case QueryType::QUERY_CREATE_TABLE: return "CREATE TABLE";
+    case QueryType::QUERY_CREATE_TRIGGER: return "CREATE TRIGGER";
+    case QueryType::QUERY_DROP:return "DROP";
+    case QueryType::QUERY_INSERT:return "INSERT";
+    case QueryType::QUERY_SET: return "SET";
+    case QueryType::QUERY_SHOW: return "SHOW";
+    case QueryType::QUERY_UPDATE: return "UPDATE";
+    case QueryType::QUERY_ALTER: return "ALTER";
+    case QueryType::QUERY_DELETE: return "DELETE";
+    case QueryType::QUERY_COPY: return "COPY";
+    case QueryType::QUERY_ANALYZE: return "ANALYZE";
+    case QueryType::QUERY_RENAME: return "RENAME";
+    case QueryType::QUERY_PREPARE: return "PREPARE";
+    case QueryType::QUERY_EXECUTE: return "EXECUTE";
+    case QueryType::QUERY_SELECT: return "SELECT";
+    case QueryType::QUERY_OTHER: default: return "OTHER";
+  }
+}
+
+QueryType StringToQueryType(std::string str) {
```

Comment:
Since we only use the value of the string, it would be better if we use a `const std::string *` as the parameter to avoid copying. Also, seems like we will construct the hash map each time we call this function.

Reply:
Fixed